src/xpring-client.ts

```diff
@@ -18,11 +19,17 @@ class XpringClient {
    * The XpringClient will use gRPC to communicate with the given endpoint.
    *
    * @param grpcURL The URL of the gRPC instance to connect to.
+   * @param useNewProtocolBuffers If `true`, then the new protocol buffer implementation from rippled will be used. Defaults to false.
    */
-  public constructor(grpcURL: string) {
-    const defaultXpringClient = LegacyDefaultXpringClient.defaultXpringClientWithEndpoint(
+  public constructor(grpcURL: string, useNewProtocolBuffers: boolean = false) {
+    var defaultXpringClient: XpringClientDecorator = LegacyDefaultXpringClient.defaultXpringClientWithEndpoint(
       grpcURL
     );
+    if (useNewProtocolBuffers) {
+      defaultXpringClient = DefaultXpringClient.defaultXpringClientWithEndpoint(
```

Comment:
So if using the new proto buffers we initialize to the Legacy and then change it to the rippled client? It might be better practice to initialize the defaultXpringClient only once.

Reply:
I changed this to using a ternary operator. It lets us use a `const` but I think it's slightly more readable. I don't care because it is functionally the same code and will go away eventually so just LMK if you prefer I revert.

Reply:
@keefertaylor Looks :fire: now!